sdk/core/Azure.Core/src/Shared/AutoRest/ManagementPipelineBuilder.cs

```diff
@@ -14,9 +14,9 @@ namespace Azure.Core
 {
     internal static class ManagementPipelineBuilder
     {
-        public static HttpPipeline Build(TokenCredential credential, ClientOptions options)
+        public static HttpPipeline Build(TokenCredential credential, string host, ClientOptions options)
         {
-            return HttpPipelineBuilder.Build(options, new BearerTokenAuthenticationPolicy(credential, "https://management.azure.com//.default"));
+            return HttpPipelineBuilder.Build(options, new BearerTokenAuthenticationPolicy(credential, $"{host}/.default"));
```

Comment:
Nit: string formatting is more expensive than just string concat. If this wasn't runtime code I wouldn't really say anything, but we could shave off some time/memory here.

Reply:
IL produced in both cases is the same (a call to string.Concat(string, string)):

https://sharplab.io/#v2:C4LglgNgNAJiDUAfAAgJgIwFgBQyDMABGgQMIEDeOB1RhyALAQLIAUAysAE5gB2A5gQAWAewDOwAJQUqNWQDcAhpwIKCAXgIASAETkR4gL4B6AHQwApgDMFAVwjBtAbhmzqi5QCN1QscALwCbVMLazsHZ2xXGhcCAxwDIA==